test/test_eval.py

```diff
@@ -59,3 +63,51 @@ def test_f1_score_at_k(self):
                                            user_features=self.user_features,
                                            item_features=self.item_features,
                                            preserve_rows=True))
+
+    def test_ndcg_at_k(self):
+
+        ndcg10 = normalized_discounted_cumulative_gain(model=self.model,
+                                                       test_interactions=self.interactions,
+                                                       k=10,
+                                                       user_features=self.user_features,
+                                                       item_features=self.item_features,
+                                                       preserve_rows=False)
+
+        ndcg20 = normalized_discounted_cumulative_gain(model=self.model,
+                                                       test_interactions=self.interactions,
+                                                       k=20,
+                                                       user_features=self.user_features,
+                                                       item_features=self.item_features,
+                                                       preserve_rows=False)
+
+        ndcg5 = normalized_discounted_cumulative_gain(model=self.model,
+                                                      test_interactions=self.interactions,
+                                                      k=5,
+                                                      user_features=self.user_features,
+                                                      item_features=self.item_features,
+                                                      preserve_rows=False)
+
+        self.assertIsNotNone(ndcg10)
+        self.assertGreaterEqual(ndcg10, ndcg5)
+        self.assertEqual(ndcg20, ndcg10)
+        self.assertLess(ndcg20, 1)
+
+    def test_idcg_at_k(self):
+
+        ordered = np.array([3, 3, 3, 2, 2, 2, 1, 0])
+        wiki_idcg = np.array([3, 3, 3, 2, 2, 2, 1, 0])
```

Comment:
http://people.cs.georgetown.edu/~nazli/classes/ir-Slides/Evaluation-13.pdf

These slides have some examples with expected values for DCG/IDCG/nDCG. Maybe they're more applicable than the wiki results?